protocol/v2/qbft/controller/controller.go

```diff
@@ -194,14 +189,17 @@ func (c *Controller) GetIdentifier() []byte {
 	return c.Identifier
 }
 
-// isFutureMessage returns true if message height is from a future instance.
-// It takes into consideration a special case where FirstHeight didn't start but  c.Height == FirstHeight (since we bump height on start instance)
-func (c *Controller) isFutureMessage(msg *specqbft.ProcessingMessage) (bool, error) {
+// validateMsgHeight returns error if provided message height is from a future instance.
+// It takes into consideration a special case where FirstHeight instance didn't start yet
+// but c.Height == FirstHeight.
+func (c *Controller) validateMsgHeight(msg *specqbft.ProcessingMessage) error {
 	if c.Height == specqbft.FirstHeight && c.StoredInstances.FindInstance(c.Height) == nil {
-		return true, nil
+		return fmt.Errorf("qbft instance at height: %d hasn't started yet", specqbft.FirstHeight)
 	}
-
-	return msg.QBFTMessage.Height > c.Height, nil
+	if msg.QBFTMessage.Height > c.Height {
+		return fmt.Errorf("message height: %d > current qbft height: %d", msg.QBFTMessage.Height, c.Height)
+	}
+	return nil
 }
```

Comment:
Just rewriting this func `isFutureMessage`->`validateMsgHeight` so I can provide a detailed error message on what exactly happened (expected vs actual).